### PR TITLE
docs(carousel): add demo with noPause function

### LINKED ISSUE #3645: docs(carousel): add demo for noPause function
add demo with button disable/enable hover pause(#3232)

### PR DESCRIPTION
**Pull request by tarusin**
add demo with toggle pause on hover button
related to #3645

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+carousel/carousel-section.list.ts [hunk 1]
```diff
@@ -57,6 +58,13 @@ export const demoComponentContent: ContentSection[] = [
         component: require('!!raw-loader?lang=typescript!./demos/dynamic/dynamic.ts'),
         html: require('!!raw-loader?lang=markup!./demos/dynamic/dynamic.html'),
         outlet: DemoCarouselDynamicComponent
+      },
+      {
+        title: 'No Pause Slides',
```
Comment (IlyaSurmay) [inline demo/src/app/components/+carousel/carousel-section.list.ts]: Pause on hover

File: demo/src/app/components/+carousel/demos/no-pause/no-pause.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="carousel-caption d-none d-md-block">
+      <h3>Third slide label</h3>
+      <p>Praesent commodo cursus magna, vel scelerisque nisl consectetur.</p>
+    </div>
+  </slide>
+</carousel>
+<br/>
+<div>
+  <button type="button" class="btn btn-info"
+          (click)="noPauseSlide()">Toggle pause on hover
```
Comment (IlyaSurmay) [inline demo/src/app/components/+carousel/demos/no-pause/no-pause.html]: no need for a function, change the binding here

File: demo/src/app/components/+carousel/demos/no-pause/no-pause.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  selector: 'demo-carousel-no-pause',
+  templateUrl: './no-pause.html'
+})
+export class DemoCarouselNoPauseComponent {
+  myInterval = 2500;
+  noWrapSlides = false;
+  showIndicator = true;
+  noPauseSlides = true;
+
+  noPauseSlide() {
```
Comment (IlyaSurmay) [inline demo/src/app/components/+carousel/demos/no-pause/no-pause.ts]: remove this

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+carousel/demos/no-pause/no-pause.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="carousel-caption d-none d-md-block">
+      <h3>Third slide label</h3>
+      <p>Praesent commodo cursus magna, vel scelerisque nisl consectetur.</p>
+    </div>
+  </slide>
+</carousel>
+<br/>
+<div>
+  <button type="button" class="btn btn-info"
+          (click)="this.noPauseSlides = !this.noPauseSlides">Toggle pause on hover
```
Comment (IlyaSurmay) [inline demo/src/app/components/+carousel/demos/no-pause/no-pause.html]: remove `this`

File: demo/src/app/components/+carousel/demos/no-pause/no-pause.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-carousel-no-pause',
+  templateUrl: './no-pause.html'
+})
+export class DemoCarouselNoPauseComponent {
+  myInterval = 2500;
+  noWrapSlides = false;
+  showIndicator = true;
```
Comment (IlyaSurmay) [inline demo/src/app/components/+carousel/demos/no-pause/no-pause.ts]: why do you need all these properties? leave only  `noPauseSlides` and rename it to `noPause`

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Please, update your pr from latest dev and fix e2e tests, if they continue fails

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Very hard to test and to see difference, please, set interval for slide changing smaller

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+carousel/carousel-section.list.ts [hunk 6]
```diff
@@ -2,15 +2,16 @@ import { DemoCarouseBasicComponent } from './demos/basic/basic';
...
 
 import { ContentSection } from '../../docs/models/content-section.model';
 import { DemoTopSectionComponent } from '../../docs/demo-section-components/demo-top-section/index';
 import { ExamplesComponent } from '../../docs/demo-section-components/demo-examples-section/index';
 import { ApiSectionsComponent } from '../../docs/demo-section-components/demo-api-section/index';
 
 import {
-  NgApiDocComponent,
-  NgApiDocConfigComponent
+NgApiDocComponent,
```
Comment (valorkin) [inline demo/src/app/components/+carousel/carousel-section.list.ts]: code formatting!

File: demo/src/app/components/+carousel/carousel-section.list.ts [hunk 7]
```diff
@@ -57,6 +58,13 @@ export const demoComponentContent: ContentSection[] = [
         component: require('!!raw-loader?lang=typescript!./demos/dynamic/dynamic.ts'),
         html: require('!!raw-loader?lang=markup!./demos/dynamic/dynamic.html'),
         outlet: DemoCarouselDynamicComponent
+      },
+      {
+        title: 'Pause on hover',
+        anchor: 'pause-on-hover',
+        component: require('!!raw-loader?lang=typescript!./demos/no-pause/no-pause.ts'),
+        html: require('!!raw-loader?lang=markup!./demos/no-pause/no-pause.html'),
```
Comment (valorkin) [inline demo/src/app/components/+carousel/carousel-section.list.ts]: demo description

**Review by valorkin — APPROVED** [review]

**Pull request merged**